analyzers/src/SonarAnalyzer.CSharp/Extensions/ITupleOperationWrapperExtensions.cs

```diff
@@ -0,0 +1,51 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using StyleCop.Analyzers.Lightup;
+
+namespace SonarAnalyzer.Extensions
+{
+    internal static class ITupleOperationWrapperExtensions
+    {
+        public static ImmutableArray<IOperation> AllElements(this ITupleOperationWrapper tuple)
+        {
+            var arrayBuilder = ImmutableArray.CreateBuilder<IOperation>();
+            CollectTupleElements(tuple);
+            return arrayBuilder.ToImmutableArray();
+
+            void CollectTupleElements(ITupleOperationWrapper tuple)
+            {
+                foreach (var element in tuple.Elements)
+                {
+                    if (ITupleOperationWrapper.IsInstance(element))
+                    {
+                        CollectTupleElements(ITupleOperationWrapper.FromOperation(element));
```

Comment:
[! recursion alert]

We've previously had some problems (sample: https://github.com/SonarSource/sonar-dotnet/issues/2429) with large initializations (see repro https://github.com/costin-zaharia-sonarsource/repro-2429/blob/main/Repro2429/DictionaryInitialization.cs) on dictionaries.

It might not be the case for LVA (DeadStores) , but this extension method will be able to be used in the future for other tuple operations in other context. I can imagine some generated code using huge nested tuples - can you do a stress test for that scenario, to ensure this method won't be a performance hotspot?

Reply:
The other PR is about a collection initializer that indeed can have thousands of entries. This doesn't apply to tuples in C# (other languages use tuples and list interchangeably so it may apply there). Further, there may only be a problem with deep nesting, as copying the elements piece by piece is a pretty standard operation. I have seen much worse things happening in the code base than this. Even further, I can't find any `[Benchamrk]` tests in the code base. How do we perform performance tests?

Reply:
see `UnusedPrivateMember_Performance` in `sonar-dotnet\analyzers\tests\SonarAnalyzer.UnitTest\Rules\UnusedPrivateMemberTest.cs`